sdk/storage/Azure.Storage.Files.Shares/tests/PartitionedDownloaderTests.cs

```diff
@@ -0,0 +1,288 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core.TestFramework;
+using Azure.Storage.Files.Shares.Models;
+using Moq;
+using NUnit.Framework;
+
+namespace Azure.Storage.Files.Shares.Tests
+{
+    [TestFixture(true)]
+    [TestFixture(false)]
+    public class PartitionedDownloaderTests
```

Comment:
we should also add tests for sharefileclient. lease take a look at `DownloadTo` (and other) tests for blobs https://github.com/Azure/azure-sdk-for-net/blob/5e6bcc0dd446cd0cbe204c00d190ef655e9d9c96/sdk/storage/Azure.Storage.Blobs/tests/BlobBaseClientTests.cs#L1473 .